src/pvrclient-nextpvr.cpp

```diff
@@ -285,17 +285,31 @@ bool cPVRClientNextPVR::Connect()
                   {
                     if ( g_livestreamingmethod != RealTime)
                     {
-                      if (version >= 50001)
+                      if (g_livestreamingmethod == Transcoded)
+                      {
+                        delete m_timeshiftBuffer;
+                        m_timeshiftBuffer = new timeshift::TranscodedBuffer();
```

Comment:
is it guaranteed that m_timeshiftBuffer will not leak?

Reply:
Again this goes back to the 2018 class design, with the application never anticipating a null return from any buffer constructor.  So far this hasn't been a reported issue,  but I will test this when these buffers are first called in OpenLiveStream() for existing code and GetStreamProperties() for this PR

Reply:
For me this looks like a memory leak. Allocation on connect. Nothing on disconnect.

Reply:
The default base classes are instantiated in the constructor and what your are seeing is a class morphing into a new type from the default.  There is a leak in that they aren't deleted in the destructor but the addon code tells the user to restart anyway.  I will make a change to delete them.

Reply:
Classes are now deleted in the Destructor()